src/Microsoft.AspNet.Authentication/Events/SigningInContext.cs

```diff
@@ -25,6 +28,13 @@ public abstract class ReturnEndpointContext : EndpointContext
         public ClaimsPrincipal Principal { get; set; }
         public AuthenticationProperties Properties { get; set; }
 
+        public bool IsRequestCompleted { get; private set; }
+
+        public void RequestCompleted()
+        {
+            IsRequestCompleted = true;
```

Comment:
It's doing very much the same thing as BaseControlContext.HandleResponse() https://github.com/aspnet/Security/blob/dev/src/Microsoft.AspNet.Authentication/Events/BaseControlContext.cs#L31